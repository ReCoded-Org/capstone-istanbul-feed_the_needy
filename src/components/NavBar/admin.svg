<svg width="32" height="34" viewBox="0 0 32 34" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M15 29.3868C21.5799 29.3868 27 23.7974 27 17.0118C27 10.2263 21.5684 4.625 15 4.625C8.4201 4.625 3 10.2263 3 17.0118C3 23.7974 8.4201 29.3868 15 29.3868ZM21.7522 23.6316C20.5005 22.2697 18.1005 21.1447 15 21.1447C11.911 21.1447 9.511 22.2697 8.25933 23.6316C6.67464 21.9145 5.71005 19.5816 5.71005 17.0118C5.71005 11.6947 9.83254 7.4079 15 7.4079C20.156 7.4079 24.3014 11.6947 24.3129 17.0118C24.3129 19.5816 23.3368 21.9145 21.7522 23.6316ZM15 19.2618C17.2392 19.2737 18.9732 17.3079 18.9732 14.7737C18.9732 12.3816 17.2163 10.3684 15 10.3684C12.7837 10.3684 11.0153 12.3816 11.0383 14.7737C11.0383 17.3079 12.7722 19.25 15 19.2618Z" fill="#F5F6F6"/>
</svg>