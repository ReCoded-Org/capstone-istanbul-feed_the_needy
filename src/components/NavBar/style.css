.logo {
  width: 50px;
  height: 31px;
  margin: 4px 16px 16px 24px;
  float: left;
}
.title {
  height: 31px;
  margin: 8px 80px 16px 16px;
  float: left;
}
.cart {
  width: 50px;
  margin: 0px 80px 0px 0px;
  float: right;
}
.admin {
  width: 50px;
  float: right;
}
.header {
  background-color: #f0b32c;
}
.stacked-title {
  width: 50px;
  float: left;
  display: none;
  margin-right: 16px;
  height: 10%;
  transform: scale(1.5);
}

.ant-menu.ant-menu-dark .ant-menu-item-selected {
  background-color: #bf5f06; /*Overriden property*/
  opacity: 0.6;
}
.ant-menu.ant-menu-dark .ant-menu-item:hover {
  background-color: #bf5f06; /*Overriden property*/
  opacity: 0.6;
}

.ant-menu.ant-menu-dark,
.ant-menu-dark .ant-menu-sub,
.ant-menu.ant-menu-dark .ant-menu-sub {
  background: #f0b32c;
}

.ant-menu-dark.ant-menu-dark:not(.ant-menu-horizontal) .ant-menu-item-selected {
  background: #bf5f06;
}

@media only screen and (max-width: 576px) {
  /*mobile*/
  .cart {
    float: none;
    margin: 0px;
  }
  .admin {
    float: none;
  }
  .logo {
    margin: 4px 16px 16px 0px;
  }
  .title {
    display: none;
  }
  .stacked-title {
    display: none;
  }
}
@media only screen and (max-width: 768px) and (min-width: 576px) {
  /*ipad*/
  .title {
    display: none;
  }
  .stacked-title {
    display: initial;
    margin: 0px 24px 0px 16px;
  }
}

@media only screen and (max-width: 1024px) and (min-width: 768px) {
  /*ipad pro*/

  .title {
    transform: scale(0.8);
    margin-right: auto;
  }
}
