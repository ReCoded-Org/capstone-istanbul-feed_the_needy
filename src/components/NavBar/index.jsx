import React from "react"
import { Link } from "react-router-dom"
/* import {
  enable as enableDarkMode,
  disable as disableDarkMode,
} from "darkreader" */
import Icon, { GlobalOutlined } from "@ant-design/icons"
import { Layout, Menu } from "antd"
import { useTranslation } from "react-i18next"
import i18next from "i18next"
import { ReactComponent as AdminSvg } from "./images/admin.svg"
import { ReactComponent as CartSvg } from "./images/cart.svg"
import { ReactComponent as LogoSvg } from "../../images/logo.svg"
import { ReactComponent as FoodOnCouponSvg } from "../../images/foodOnCoupon.svg"
import { ReactComponent as StackedTitleSvg } from "../../images/stackedTitle.svg"
import "./style.css"

const { Header } = Layout
const { SubMenu } = Menu

const NavBar = ({ isTesting }) => {
  const { t } = useTranslation()

  return (
    <>
      <Header className="navbarHeader">
        <div className="logo">
          <LogoSvg />
        </div>
        <div className="navbarTitle">
          <FoodOnCouponSvg />
        </div>
        <div className="navbarStackedTitle">
          <StackedTitleSvg />
        </div>
        <Menu
          style={{ background: "#F0B32C" }}
          theme="dark"
          mode={isTesting ? "vertical" : "horizontal"}
          defaultSelectedKeys={["1"]}
        >
          <Menu.Item key="1">
            <Link to="/">{t("navbar.headers.header0")}</Link>
          </Menu.Item>
          <Menu.Item key="2">
            <Link to="/coupons">{t("navbar.headers.header1")}</Link>
          </Menu.Item>
          <Menu.Item key="3">
            <Link to="/blog">{t("navbar.headers.header2")}</Link>
          </Menu.Item>
          <SubMenu key="sub1" icon={<GlobalOutlined />}>
            <Menu.Item
              onClick={() => {
                i18next.changeLanguage("tr")
              }}
              key="4"
            >
              {t("navbar.languages.lang0")}
            </Menu.Item>
            <Menu.Item
              onClick={() => {
                i18next.changeLanguage("en")
              }}
              key="5"
            >
              {t("navbar.languages.lang1")}
            </Menu.Item>
            <Menu.Item
              onClick={() => {
                i18next.changeLanguage("ar")
              }}
              key="6"
            >
              {t("navbar.languages.lang2")}
            </Menu.Item>
          </SubMenu>
          <Menu.Item key="7" className="navbarCart">
            <Link to="/cart">
              <Icon style={{ transform: "scale(2)" }} component={CartSvg} />
            </Link>
          </Menu.Item>
          <Menu.Item key="8" className="navbarAdmin">
            <Link to="/admin">
              <Icon style={{ transform: "scale(2)" }} component={AdminSvg} />
            </Link>
          </Menu.Item>
          <Menu.Item
            key="9"
            className="navbarAdmin"
            onClick={() => {
              enableDarkMode()
            }}
          >
            <span role="img" aria-label="dayTheme">
              🌙
            </span>
          </Menu.Item>
          <Menu.Item
            key="10"
            className="navbarAdmin"
            onClick={() => {
              disableDarkMode()
            }}
          >
            <span role="img" aria-label="nightTheme">
              ☀️
            </span>
          </Menu.Item>
        </Menu>
      </Header>
    </>
  )
}

export default NavBar
