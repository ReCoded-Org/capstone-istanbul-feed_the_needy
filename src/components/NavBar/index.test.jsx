import React from 'react';
import renderer from 'react-test-renderer';
import NavBar from './index';

it('renders correctly', () => {
    const tree = renderer
      .create(<NavBar />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });