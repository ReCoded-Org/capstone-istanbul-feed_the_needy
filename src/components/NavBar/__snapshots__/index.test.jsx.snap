// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<header
  className="ant-layout-header header"
>
  <div
    className="logo"
  >
    <svg>
      logo.svg
    </svg>
  </div>
  <div
    className="title"
  >
    <svg>
      foodOnCoupon.svg
    </svg>
  </div>
  <div
    className="stacked-title"
  >
    <svg>
      stackedTitle.svg
    </svg>
  </div>
  <ul
    className="ant-menu ant-menu-dark ant-menu-root ant-menu-vertical"
    onMouseEnter={[Function]}
    onTransitionEnd={[Function]}
    role="menu"
    style={
      Object {
        "background": "#F0B32C",
      }
    }
  >
    <li
      className="ant-menu-item ant-menu-item-selected"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={Object {}}
    >
       
      <a
        href="/"
        onClick={[Function]}
      >
        navbar.headers.header0
      </a>
       
    </li>
    <li
      className="ant-menu-item ant-menu-item-only-child"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={Object {}}
    >
      <a
        href="/coupons"
        onClick={[Function]}
      >
        navbar.headers.header1
      </a>
    </li>
    <li
      className="ant-menu-item ant-menu-item-only-child"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={Object {}}
    >
      <a
        href="/blog"
        onClick={[Function]}
      >
        navbar.headers.header2
      </a>
    </li>
    <li
      className="ant-menu-item ant-menu-item-only-child cart"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={Object {}}
    >
      <a
        href="/cart"
        onClick={[Function]}
      >
        <span
          className="anticon"
          role="img"
          style={
            Object {
              "transform": "scale(2)",
            }
          }
        >
          <svg
            aria-hidden="true"
            className=""
            fill="currentColor"
            focusable="false"
            height="1em"
            width="1em"
          >
            cart.svg
          </svg>
        </span>
      </a>
    </li>
    <li
      className="ant-menu-item ant-menu-item-only-child admin"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={Object {}}
    >
      <a
        href="/admin"
        onClick={[Function]}
      >
         
        <span
          className="anticon"
          role="img"
          style={
            Object {
              "transform": "scale(2)",
            }
          }
        >
          <svg
            aria-hidden="true"
            className=""
            fill="currentColor"
            focusable="false"
            height="1em"
            width="1em"
          >
            admin.svg
          </svg>
        </span>
         
      </a>
    </li>
  </ul>
</header>
`;
